scylla/src/transport/host_filter.rs

```diff
@@ -54,7 +54,7 @@ impl AllowListHostFilter {
 
 impl HostFilter for AllowListHostFilter {
     fn accept(&self, peer: &Peer) -> bool {
-        self.allowed.contains(&peer.address)
+        self.allowed.contains(&peer.address.inner())
```

Comment:
Hmm, so now we are consulting the host filter for all types of addresses (RealPeer, FixedLocal, InitiallyKnown). I think that we should only ask the hashmap about RealPeer addresses and return true for others:

- If the address is InitiallyKnown, the user put the address into the initial peers list, so they most likely want to connect to this node,
- If the address is FixedLocal, then the node either was originally an InitiallyKnown node, or a RealPeer node for which the host filter returned true.

What do you think?

Reply:
Does this preserve the HostFilter's behaviour regarding initially known nodes? Did the filter accept them all previously?

Reply:
Previously, the AllowListHostFilter was able to filter out initial nodes, but the driver would still be able to establish a control connection to one of them - if it happened, it would print a warning about it and switch the control connection to another node.

It's not really possible to preserve the semantics. Previously, AllowListHostFilter was always given a translated address, and now we don't do that - and we can't because the notion of a translated address makes less and less sense: node's address can change, multiple nodes can be behind a proxy, etc.

Maybe it's better to remove the AllowListHostFilter. It's not clear what its semantics should be. It's also quite simple for users to implement something like this if they need it and it will force them to think about how to handle different address types. Anyway, the main motivation for the host filter interface was introducing the DcHostFilter, which should still work.